chapters/data-processing.tex

```diff
@@ -0,0 +1,725 @@
+but a few columns of the format \texttt{gender\_1}, \texttt{gender\_2}, etc.
+The data is saved in this format because it's the most efficient way to transfer it:
+adding different levels of observation into the same table allows for data to be transferred in a single file.
+However, this leads to the widespread practice of interacting with data in wide format for a large part of a project,
+although this is not the most efficient way to do so.
+
+To understand how dealing with wide data can be complicated,
+just suppose you need to calculate the share of women in your sample.
+The first thing you will need to do is change the format of your table so that each row represents one individual.
+Tidy datasets, on the other hand, 
+are easier to handle because observations can be aggregated and linked consistently,
+without additional steps.
+They are also easier to clean, 
+as each attribute only needs to be checked once,
+as corresponds directly to one question in the questionnaire.
+
+We have discussed only wide data so far,
+but as mentioned earlier, there are unlimited ways for data to be untidy.
+An alternative example is a table containing information on transactions
+and on the parties involved in each transaction.
+In this case, the firm-level information will be repeated for all transactions a given firm was involved in.
+Analyzing data in this format would give more weight to firms that conducted more transactions,
+which may not be consistent with your design.
+
+The basic process behind tidying a dataset is simple: 
+first, you need to identify all the variables that were measured at the same level of observation;
+then, create separate tables for each level of observation;
+finally, reshape\sidenote{\textbf{Reshape:}
+	transform a dataset in such a way that the unit of observation represented by a row changes.} 
+the data and remove duplicated rows 
+until each table is uniquely and fully identified by the ID variable 
+that corresponds to its unit of observation.
+In the earlier household survey example,
+household-level variables will be stored in one tidy dataset, 
+and household-member variables are reshaped to member level and stored in another tidy dataset,
+which also contains the household ID for each individual.
+So the household ID is intentionally duplicated in the household members table.
+In the transaction data example,
+the result of the tidying process would be one transaction-level dataset, 
+containing variables indicating the ID of all firms involved;
+and one firm-level dataset with a single entry for each firm.
+
+Reshaping tables is the most intricate task in this process,
+so you should be very familiar with commands such at \texttt{reshape} in Stata and \texttt{pivot} in R.
+Another key aspect in this process is to make sure that ID variables are consistent across tables,
+so they can always be linked.
+For each tidy table, you should have a corresponding master dataset in the same unit of observation.
```

Comment:
Should we? Like if you have a nested loop that is fertilizer per season per crop per plot, should you have a master data set for each of them? Let's discuss!

Reply:
There must be a clear way to connect it to a master dataset

Reply:
addressed in e23ce52